server/src/main/java/com/linecorp/centraldogma/server/internal/ServiceUtil.java

```diff
@@ -0,0 +1,115 @@
+
+/**
+ * Utility methods that are useful when implementing a service.
+ */
+public final class ServiceUtil {
+
+    /**
+     * Aggregates the JSON files in the specified paths and returns the {@link AggregatedEntry}.
+     */
+    public static CompletableFuture<AggregatedEntryDto<?>> aggregatedEntries(
+            Revision normalizedRevision, Repository repository,
+            Iterable<PathAndOptional> paths, Iterable<String> expressions) {
+
+        requireNonNull(normalizedRevision, "normalizedRevision");
+        requireNonNull(repository, "repositoy");
+        requireNonNull(paths, "paths");
+        requireNonNull(expressions, "expressions");
+
+        // Only JSON files can currently be aggregated.
+        paths.forEach(path -> validateJsonFilePath(path.getPath(), "path"));
+
+        final List<CompletableFuture<Entry<?>>> entryFutures = new ArrayList<>(Iterables.size(paths));
+        paths.forEach(path -> entryFutures.add(repository.get(normalizedRevision, path.getPath())));
+        final CompletableFuture<Void> allDoneFuture = CompletableFuture.allOf(
+                entryFutures.toArray(new CompletableFuture[0]));
+
+        final CompletableFuture<AggregatedEntryDto<?>> future = new CompletableFuture<>();
+        allDoneFuture.whenComplete((unused1, unused2) -> {
+            JsonNode result;
+            try {
+                result = aggregateJson(paths, entryFutures);
+            } catch (Exception e) {
+                future.completeExceptionally(e);
+                return;
+            }
+
+            if (Iterables.size(expressions) > 0) {
+                try {
+                    result = Jackson.extractTree(result, expressions);
+                } catch (Exception e) {
+                    // A user sent the request with a invalid JSON path or pointing to the property which
+                    // the aggregated JSON doesn't have.
+                    future.completeExceptionally(new IllegalArgumentException(e.getMessage()));
+                    return;
+                }
+            }
+
+            future.complete(new AggregatedEntryDto<>(EntryType.JSON, result));
+        });
+        return future;
+    }
+
+    private static JsonNode aggregateJson(Iterable<PathAndOptional> paths,
+                                          List<CompletableFuture<Entry<?>>> entryFutures) throws Exception {
+        final Builder<JsonNode> builder = new Builder<>();
+        for (int i = 0; i < entryFutures.size(); i++) {
+            try {
+                final Entry<?> entry = entryFutures.get(i).get();
+                builder.add(entry.contentAsJson());
+            } catch (Exception e) {
+                final PathAndOptional path = Iterables.get(paths, i);
+                if (!path.isOptional()) {
+                    if (e instanceof JsonParseException) {
+                        throw new CompletionException("failed to parse the content as JSON. path: " +
```

Comment:
How about using another exception? Because it might be peeled at several points in our code, so its message(ex, path) might be lost.

Reply:
Changed the exception to `IllegalStateException` because the server stores a JSON file content which is not a JSON document(which is not possible). Thank you~!